### ISSUE TITLE
Unable to paste image within fully-selected webpage content

### ISSUE THREAD

**Issue by hwangdev97** [toeverything/blocksuite#2855]
In the same article, when the entire text is selected, it is not possible to paste images, but when only part of the text is selected, it is possible to paste images: 

https://github.com/toeverything/blocksuite/assets/24713927/dccf5db0-14cc-4dbb-be0b-74842c2726dc

**Comment by doodlewind** [comment]
@hwangdev97 could you provide a link to the test page as the input source?

**Comment by hwangdev97** [comment]
here: https://sspai.com/post/62414

**Comment by youhua** [comment]
I have determined that the bug is caused by the presence of Markdown syntax in the page's code. This leads to the conversion of pasted content into Markdown format, causing images to be lost.

BTW, I have discovered a related issue where pasting the same content twice results in different outcomes. The second paste displays the image and style correctly. This issue is caused by the regular expression that recognizes markdown syntax, which has the /g flag.

I am planning to fix this issue. May I know if anyone else is already working on it?
